src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -447,9 +448,19 @@ void VST3Wrapper::FetchSettings(EffectSettings& settings)
       auto cleanup = finally([&] { componentHandler->EndStateChange(); });
 
       //Restore state
-      const auto* vst3settings = &GetSettings(settings);
-      if(!vst3settings->processorState.has_value())
-         vst3settings = &GetSettings(mDefaultSettings);
+      // See Issue 3942
+      // Callbacks to performEdit might happen inside setState that call
+      // ModifySettings to reassign the given settings, but the settings may be
```

Comment:
`BeginStateChange` prevents `ModifySettings` from being called in the same callstack with `setState`

Reply:
You are right.

Do you recall why you wrote BeginStateChange?

Were you observing exactly the sort of dangling reference problem I destribed?